x-pack/plugins/apm/public/components/app/ServiceDetails/ServiceIntegrations/MachineLearningFlyout.tsx

```diff
@@ -0,0 +1,280 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiButton,
+  EuiCallOut,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFlyout,
+  EuiFlyoutBody,
+  EuiFlyoutFooter,
+  EuiFlyoutHeader,
+  EuiFormRow,
+  EuiSpacer,
+  EuiText,
+  EuiTitle
+} from '@elastic/eui';
+import React, { Component } from 'react';
+import { toastNotifications } from 'ui/notify';
+import {
+  getMlPrefix,
+  startMLJob
+} from 'x-pack/plugins/apm/public/services/rest/ml';
+import {
+  getAPMIndexPattern,
+  ISavedObject
+} from 'x-pack/plugins/apm/public/services/rest/savedObjects';
+import { MLJobsRequest } from 'x-pack/plugins/apm/public/store/reactReduxRequest/machineLearningJobs';
+import { KibanaLink, ViewMLJob } from 'x-pack/plugins/apm/public/utils/url';
+import { TransactionSelect } from './TransactionSelect';
+
+interface FlyoutProps {
+  isOpen: boolean;
+  onClose: () => void;
+  serviceName: string;
+  transactionType?: string;
+  location: any;
+  serviceTransactionTypes: string[];
+}
+
+interface FlyoutState {
+  isLoading: boolean;
+  hasMLJob: boolean;
+  hasIndexPattern: boolean;
+  selectedTransactionType: string;
+}
+
+export class MachineLearningFlyout extends Component<FlyoutProps, FlyoutState> {
+  public state = {
+    isLoading: false,
+    hasIndexPattern: false,
+    hasMLJob: false,
+    selectedTransactionType:
+      this.props.transactionType || this.props.serviceTransactionTypes[0]
+  };
+
+  public async componentDidMount() {
+    const indexPattern: ISavedObject | undefined = await getAPMIndexPattern();
+    this.setState({ hasIndexPattern: !!indexPattern });
+  }
+
+  public componentDidUpdate(prevProps: FlyoutProps) {
+    if (prevProps.transactionType !== this.props.transactionType) {
+      this.setState({
+        selectedTransactionType:
+          this.props.transactionType || this.props.serviceTransactionTypes[0]
+      });
+    }
+  }
```

Comment:
I can only change this value if transaction type changes, otherwise I want to use the value in state that is otherwise controlled by the select box. I'm sure there's a better way to do this but I'm not seeing it yet, punting for now.

Reply:
Sounds good. Let's pick the importing battles for now. 👍